src/common/moisture_sensor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::common::analog::AnalogReader;
+use crate::common::sensor::GenericReadingsResult;
+use crate::common::sensor::Sensor;
+use crate::common::sensor::SensorResult;
+use crate::common::sensor::SensorT;
+use crate::common::sensor::TypedReadingsResult;
+use crate::common::status::Status;
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+pub struct MoistureSensor {
+    analog: Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>,
+}
+
+impl MoistureSensor {
+    pub fn new(analog: Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>) -> Self {
+        MoistureSensor { analog: analog }
```

Comment:
you can write this as MoistureSensor {analog}

Reply:
Done